fastparquet/converted_types.py

```diff
@@ -174,10 +166,7 @@ def from_bytes(d):
     return data
 
 
-@numba.njit(nogil=True)
-def time_shift(indata, outdata, factor=1000):  # pragma: no cover
-    for i in range(len(indata)):
-        if indata[i] == nat:
-            outdata[i] = nat
-        else:
-            outdata[i] = indata[i] * factor
+def time_shift(indata, factor=1000):
+    temp = indata.astype('int64')
+    return (temp==nat)*nat | temp * factor
+
```

Comment:
... and use numpy's vector operators; instead of loops.

Reply:
There was a reason to use numba looping here :( 
You can run the benchmark functions in https://github.com/dask/fastparquet/blob/master/fastparquet/benchmarks/columns.py and see what difference it makes (presumably on win64).